packages/core/src/util/txInspector.ts

```diff
@@ -0,0 +1,145 @@
+import {
+  Address,
+  CertificateType,
+  Ed25519KeyHash,
+  Lovelace,
+  PoolId,
+  StakeAddressCertificate,
+  StakeDelegationCertificate,
+  TokenMap,
+  TxAlonzo,
+  TxOut,
+  Value
+} from '../Cardano';
+
+const isMyAddress =
+  (addresses: Address[]) =>
+  ({ address }: { address: Address }) =>
+    addresses.includes(address!);
+
+const isOutgoing = (tx: TxAlonzo, ownAddresses: Address[]): boolean => tx.body.inputs.some(isMyAddress(ownAddresses));
+
+const getTotalValue = (outputs: TxOut[]): Value => {
+  const result: Value = { coins: 0n };
+  const totalAssets: TokenMap = new Map();
+
+  for (const output of outputs) {
+    result.coins += output.value.coins;
+    if (output.value.assets) {
+      for (const [asset, amount] of output.value.assets) {
+        const total = totalAssets.get(asset) ?? 0n;
+        totalAssets.set(asset, total + amount);
+      }
+    }
+  }
+
+  if (totalAssets.size > 0) result.assets = totalAssets;
+  return result;
+};
+
+type InspectionResult = { [k: string]: unknown } | null;
+type InspectTx = (tx: TxAlonzo) => InspectionResult;
+type Inspectors = { [k: string]: InspectTx };
+type TxInspector<T extends Inspectors> = (tx: TxAlonzo) => {
+  [k in keyof T]: ReturnType<T[k]>;
+};
+
+export type SendReceiveCoinInspector = {
+  coins: Lovelace;
+  assets?: TokenMap;
+} | null;
+
+export type DelegateInspector = {
+  stakePool: PoolId;
+  stakeKeyHash: Ed25519KeyHash;
+} | null;
+
+export type StakeKeyRegistrationInspector = {
+  stakeKeyHash: Ed25519KeyHash;
+} | null;
+
+/**
+ * Inspects a transaction for coins and assets sent from the provided addresses.
+ * Returns `null` if it's not sending anything.
+ *
+ * @param {Address[]} ownAddresses own wallet's addresses
+ */
+export const sendCoinInspector =
+  (ownAddresses: Address[]) =>
+  (tx: TxAlonzo): SendReceiveCoinInspector => {
+    if (!isOutgoing(tx, ownAddresses)) return null;
+    const sentOutputs = tx.body.outputs.filter((out) => !isMyAddress(ownAddresses)(out));
+    return getTotalValue(sentOutputs);
+  };
+
+/**
+ * Inspects a transaction for coins and assets received by the provided addresses.
+ * Returns `null` if it's not receiving anything.
+ *
+ * @param {Address[]} ownAddresses own wallet's addresses
+ */
+export const receiveCoinInspector =
+  (ownAddresses: Address[]) =>
+  (tx: TxAlonzo): SendReceiveCoinInspector => {
+    if (isOutgoing(tx, ownAddresses)) return null;
+    const receivedOutputs = tx.body.outputs.filter((out) => isMyAddress(ownAddresses)(out));
+    return getTotalValue(receivedOutputs);
+  };
+
+/**
+ * Inspects a transaction for a stake delegation certificate.
+ * Returns `null` if the transaction doesn't have one.
```

Comment:
This looks nicer in generated docs
```suggestion
 * @returns `null` if the transaction doesn't have one.
```